ml/src/train_fn.py

```
@@ -0,0 +1,84 @@
+    criterion = getattr(torch.nn, config.loss)().to(device)
+    optimizer = getattr(torch.optim, config.optimizer.type)(model.parameters(), **config.optimizer.args)
+    scheduler = getattr(torch.optim.lr_scheduler, config.lr_scheduler.type)(optimizer, **config.lr_scheduler.args)
+
+    metrics = [getattr(module_metric, met) for met in config.metrics]
+    tracker = MetricTracker('loss', *config.metrics)
+
+    for epoch in range(1, config.train.epochs + 1):
+        model.train()
+        tracker.reset()
+
+        for batch_idx, ((x_wide, x_deep), target) in enumerate(train_dataloader):
```

Ruff (0.12.2)

Loop control variable `batch_idx` not used within loop body

Rename unused `batch_idx` to `_batch_idx`

(B007)